### PR TITLE
[CAZ-1219] Add company email and password page

### PR DESCRIPTION
**Pull request by likeuwill**
## Motivation and Context

[JIRA](https://eaflood.atlassian.net/browse/CAZ-1219)
## Checklist:

- [x] It contains only changes required by issue (does not contain other PR)
- [x] Includes link to a issue (if apply)
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: app/controllers/organisations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  #
+  # ==== Path
+  #
+  #    POST /fleets/organisation-account/create-account-name
+  #
+  # ==== Params
+  # * +name+ - account name
+  #
+  def create_account
+    CreateAccountService.call(user_params: user_params)
```
Comment (miqs1992) [inline app/controllers/organisations_controller.rb]: we need to also pass company_name

File: app/controllers/organisations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  # ==== Path
+  #
+  #    POST /fleets/organisation-account/create-account-name
+  #
+  # ==== Params
+  # * +name+ - account name
+  #
+  def create_name
+    if company_params['name'].present?
+      redirect_to email_address_and_password_path
```
Comment (miqs1992) [inline app/controllers/organisations_controller.rb]: save company_name in the session

File: app/controllers/organisations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  end
+
+  ##
+  # Renders the new email and password page.
+  #
+  # ==== Path
+  #
+  #    GET /fleets/organisation-account/email-address-and-password
+  #
+  def new_email_and_password
```
Comment (miqs1992) [inline app/controllers/organisations_controller.rb]: check if company name is present

File: app/controllers/organisations_controller.rb [hunk 4]
```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+class OrganisationsController < ApplicationController
+  ##
```
Comment (miqs1992) [inline app/controllers/organisations_controller.rb]: skip_before_action :authenticate_user!

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -4,5 +4,5 @@ class ApplicationController < ActionController::Base
   # protects applications against CSRF
   protect_from_forgery prepend: true
   # checks if a user is logged in
-  before_action :authenticate_user!
+  # before_action :authenticate_user!
```
Comment (miqs1992) [inline app/controllers/application_controller.rb]: This should be here

File: app/exceptions/new_password_exception.rb [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  # Attribute used internally
+  attr_reader :errors_object
+  #
+  # Initializer method for the class. Calls +super+ method on parent class (ApplicationException).
+  #
+  # ==== Attributes
+  # * +errors_array+ - hash - messaged passed to parent exception
+  def initialize(errors)
+    @errors_object = errors
+    super(errors_object || 'New email and password is invalid')
```
Comment (miqs1992) [inline app/exceptions/new_password_exception.rb]: I think it accepts only string - errors_object is an object

File: app/models/email_and_password_form.rb [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  # allow using ActiveRecord validation
+  include ActiveModel::Validations
+
+  # Attribute used internally
+  attr_accessor :email, :email_confirmation, :password, :password_confirmation
+
+  # rubocop:disable Style/FormatStringToken:
+  # validates attributes to presence
+  validates :email, :email_confirmation, :password, :password_confirmation,
+            presence: { message: '%{attribute} is required' }
```
Comment (miqs1992) [inline app/models/email_and_password_form.rb]: Add [TBA] to those errors (To be adjusted) and please move them to locales

File: app/models/user.rb [hunk 8]
```diff
@@ -16,7 +16,7 @@ class User
   devise :timeoutable
 
   # Attribute that is being used to authorize a user and use it in csv uploading.
-  attr_accessor :email, :admin, :sub
+  attr_accessor :email, :password, :password_confirmation, :admin, :sub
```
Comment (miqs1992) [inline app/models/user.rb]: we should not save passwords in the session, but company name might be useful

File: app/exceptions/new_password_exception.rb (hunk 6 above)
Comment (likeuwill) [reply app/exceptions/new_password_exception.rb]:
```
{:email=>["Email is required", "Email is in an invalid format", "Email address and email address confirmation must be the same"], :password_confirmation=>["Password confirmation is required", "Password confirmation must be at least 8 characters long, include at least one upper case\n      letter, a number and a special character", "Password and password confirmation must be the same"], :password=>["Password and password confirmation must be the same"], :email_confirmation=>["Email address and email address confirmation must be the same"]}
```

File: app/controllers/organisations_controller.rb [hunk 9]
```diff
@@ -26,10 +29,12 @@ def new_name
   # * +name+ - account name
   #
   def create_name
-    if company_params['name'].present?
+    form = CompanyNameForm.new(company_params)
+    if form.valid?
+      session[:company_name] = company_params['company_name']
```
Comment (miqs1992) [inline app/controllers/organisations_controller.rb]: session[:company_name] = form.company_name

File: app/controllers/organisations_controller.rb [hunk 10]
```diff
@@ -26,10 +29,12 @@ def new_name
...
       redirect_to email_address_and_password_path
     else
-      redirect_to create_account_name_path, alert: 'Enter the company name'
+      redirect_to create_account_name_path, alert: form.errors.full_messages.join(',')
```
Comment (miqs1992) [inline app/controllers/organisations_controller.rb]: Maybe use render with form.errors?

File: app/services/create_account_service.rb [hunk 11]
```diff
@@ -5,15 +5,16 @@
...
+  attr_reader :user_params, :company_name
   ##
   # Initializer method.
   #
   # ==== Attributes
   #
   # * +user_params+ - hash
-  def initialize(user_params:)
+  def initialize(user_params:, company_name:)
     @user_params = user_params
```
Comment (miqs1992) [inline app/services/create_account_service.rb]: maybe assign email nad password instead of user_params

**Pull request merged**